### PR TITLE
Add coal with carbon capture

### PR DESCRIPTION
**Pull request by ebbekyhl**
Allow coal power plant with carbon capture.

Carbon capture capital cost is based on the "biomass CHP capture" from the PyPSA/technology-data repository.

### CODE REVIEW

File: config.default.yaml [hunk 1]
```diff
@@ -232,6 +232,7 @@ sector:
   marginal_cost_storage: 0. #1e-4
   methanation: true
   helmeth: true
+  coal_cc: true
```
Comment (fneum) [inline config.default.yaml:235]:
```suggestion
  coal_cc: false
```
Let's disable this by default as the default does not feature coal power plants without carbon capture.

File: scripts/prepare_sector_network.py [hunk 2]
```diff
@@ -1118,6 +1118,24 @@ def add_storage_and_grids(n, costs):
...
+    bus3="co2 stored",
+    marginal_cost=costs.at['coal', 'efficiency'] * costs.at['coal', 'VOM'], #NB: VOM is per MWel
+    capital_cost=costs.at['coal', 'efficiency'] * costs.at['coal', 'fixed'] + costs.at['biomass CHP capture', 'fixed'] * costs.at['coal', 'CO2 intensity'], #NB: fixed cost is per MWel
+    p_nom_extendable=True,
+    carrier="coal",
+    efficiency=costs.at['coal', 'efficiency'],
+    efficiency2=costs.at['coal', 'CO2 intensity'] * (1 - costs.at['biomass CHP capture','capture_rate']),
+    efficiency3=costs.at['coal', 'CO2 intensity'] * costs.at['biomass CHP capture','capture_rate'],
+    lifetime=costs.at['coal','lifetime']
+)
```
Comment (fneum) [inline scripts/prepare_sector_network.py:1138]: This segment needs to be indented by two layers.

**Review by fneum — CHANGES_REQUESTED** [review]
Just some formatting left

File: scripts/prepare_sector_network.py (hunk 2 above)
Comment (ebbekyhl) [reply scripts/prepare_sector_network.py]: Oh yeah, indeed it does. Just did.

**Review by fneum — APPROVED** [review]

**Pull request merged**